web/src/main/java/org/apache/shiro/web/session/HttpServletSession.java

```diff
@@ -83,7 +83,8 @@ public long getTimeout() throws InvalidSessionException {
 
     public void setTimeout(long maxIdleTimeInMillis) throws InvalidSessionException {
         try {
-            int timeout = Long.valueOf(maxIdleTimeInMillis / 1000).intValue();
+        	long ltimeout = maxIdleTimeInMillis / 1000;
+        	int timeout = (int)ltimeout; 
```

Comment:
Thanks @pitjazz 
This could probably just be something like:
```java
int timeout = (int) (maxIdleTimeInMillis / 1000);
```

Can also clean up the commit and PR comment.  This isn't really a "bug" it's more code cleanup, maybe something like: 
> Removed unnecessary boxing in HttpServletSession

Reply:
SonarCloud link for this bug/cleanup: [here](https://sonarcloud.io/project/issues?id=apache_shiro&languages=java&open=AXTvO9sg3S_U1xqKcOjR&resolved=false&severities=MINOR&types=BUG).

I'll clean the commit message etc.

Reply:
I appreciate the SonarCloud link, but we need to embed the info in the commit message (the SonarCloud content could be removed at some point in the future, so we would need to provide the same context in the commit message directly.

Reply:
Thank you, I understand. Changed the commit message to suggested one. Squashed the commits to one commit.

Please review @bdemers.